hooks/useCustomStyle.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+function isMobile() {
+  const userAgent = navigator.userAgent || navigator.vendor ;
+  return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
+}
+
+// const isMobile = () => /iPhone|Android/i.test(globalThis.navigator.userAgent);
+
+function getPadding(mainElement: HTMLElement) {
+  return Number(getComputedStyle(mainElement).paddingLeft.replace(/[^0-9]/g, "")) ;
+}
+
+function getPseudoElementWidth(currentElement: HTMLElement, pseudoName : string) {
+
+    const computedStyle = getComputedStyle(currentElement, pseudoName);
+
+    // dynamically create element because of computing content width dynamically
+    const customCssElement = document.createElement('div');
+    customCssElement.style.position = 'absolute';
+    customCssElement.style.visibility = 'hidden';
+    customCssElement.style.pointerEvents = 'none';
+    customCssElement.className = 'invisible-div';
```

Comment:
className is not needed. will delete this.
.visibility = 'hidden'; is  for not showing this element because this element is only for computing width of element
.pointerEvents = 'none'; is for preventing the shape of the cursor changed